### ISSUE TITLE
Install fails on Apple M1 processors

### ISSUE THREAD

**Issue by jzalger** [particle-iot/particle-cli#611]
## Description

Installer does not recognize the new arm architecture for the Apple M1

### Steps to reproduce

When using the bash command line install on an M1 Mac

`bash <( curl -sL https://particle.io/install-cli )`

get the following traceback:

```
:::: Installing the Particle CLI for darwin to "/Users/jonathanzalger/bin/particle"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
KeyError: 'arm'
gunzip: (stdin): unexpected end of file
```

### Expected result

Install should proceed.

### Actual result

Install fails with above traceback.

### Environment

Tell us about your setup:

* OS: Mac OS 11.3 with M1 Chip
* Node (run `node -v`): v16.1.0
* NPM (run `npm -v`): 7.11.2
* Particle CLI (run `particle version`):

**Comment by busticated** [comment]
can you try again with your terminal's "open using rosetta" option checked?


![open using rosetta](https://aws1.discourse-cdn.com/business7/uploads/particle/original/3X/d/f/dfcd5432e7bef4c14c2309823e1be21be2737606.jpeg)

**Comment by jzalger** [comment]
That did resolve it. Any plans to have a build for the M1 ?
Can I help at all?

**Comment by busticated** [comment]
> Any plans to have a build for the M1?

yes, in time. we're basically downstream of Node.js (which [added support](https://github.com/nodejs/node/issues/37309) as of `v16`), `node-usb` (which seems to have [added support](https://github.com/tessel/node-usb/releases/tag/v1.7.1) as of `v1.7.1`), and `node-serialport` (where things seem a bit more [complicated](https://github.com/serialport/node-serialport/issues/2243)) so it may be a while.

> Can I help at all?

thanks for the offer 🙏  i'd point you at those upstream project(s) as they're far more in need of help than us in some regards. ultimately once our dependencies have full and stable support, we'll need to update to the latest / greatest over here, test, and release.

**Comment by edalquist** [comment]
Any updates here? I tried installing using rosetta but particle-cli depends on brew and I already have a LARGE brew install of M1 compiled dependencies
